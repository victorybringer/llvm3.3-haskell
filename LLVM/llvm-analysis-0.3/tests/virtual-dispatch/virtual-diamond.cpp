class Base {
public:
  Base() {}
  virtual void f1();
};

class Middle1 : public virtual Base {
public:
  virtual void m1() {}
  virtual void f1();
};

class Middle2 : public virtual Base {
public:
  Middle2();
  virtual void m2() const {}
};


class Derived : public Middle1, public Middle2 {
  int m_i;
public:
  Derived(int);
  virtual int f3() const;
  virtual void f1();
};

Derived::Derived(int i): m_i(i) {}
int Derived::f3() const {
//  this->m2();
  return m_i;
}



Middle2::Middle2() {

}

void Base::f1 () {

}

void Middle1::f1() {

}

void Derived::f1() {

}

void mm(Base * b) {
  b->f1();
}
