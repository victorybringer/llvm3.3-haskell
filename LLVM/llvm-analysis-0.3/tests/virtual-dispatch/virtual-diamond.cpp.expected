fromList [ ("mm", fromList ["Base::f1", "Middle1::f1", "Derived::f1"]) ]
